src/main/java/com/vmware/vipclient/i18n/messages/service/ProductService.java

```diff
@@ -72,44 +73,75 @@ public ProductService(BaseDTO dto) {
         return components;
     }
 
+    private Set<String> getSupportedLocales() {
+        return getSupportedLocales(true);
+    }
+
+    public Set<String> getCachedSupportedLocales() {
```

Comment:
There are 3 'getSupportedLocales' methods and 2 'getCachedSupportedLocales' methods, and they call each other, and seems 2 'getCachedSupportedLocales' are not the relation of overload, it's a little difficult to understand the relationship of these methods, it's better optimize the methods' name.

Reply:
I have renamed 1 method only. The overloading should be clear. Look how the methods names are self-explanatory.. descriptions seem to be repetitive of what is already obvious.

1. getSupportedLocales() - gets all supported locales (from all data sources)
2. getSupportedLocales(dataSource) - gets supported locales from the specified data source
3. getCachedSupportedLocales() - gets all supported locales (from all data sources) from cache
4. getCachedSupportedLocales(dataSource) - gets supported locales from the specified data source from cache
5. combineSupportedLocales (refreshCache) - combines the list of supported locales from all dataSources. Used in # 1 and # 3.

Reply:
according OO thinking, the dataSource should be responsible of its methods instead of product service.